### ISSUE TITLE
Broken test in _shared, interpolation constant mode

### ISSUE THREAD

**Issue by sciunto** [scikit-image/scikit-image#2233]
The test is `test_coord_map()` in `skimage/_shared/tests/test_interpolation.py`, with the `Constant` mode

What's expected in the test: 

```
expected_constant = [0, 0, 0, 0, 0, 0, 0, 1, 2, 3, 0, 0]
```

What's actually returned:

```
[-6, -5, -4, -3, -2, -1, 0, 1, 2, 3, 4, 5]
```

See #2229 (epecially @ahojnnes comments) and #2127 (tests in `_shared` are not executed)

**Comment by soupault** [comment]
Can we just remove this test?
```
    constant = [coord_map_py(4, n, 'C') for n in range(-6, 6)]
    expected_constant = [0, 0, 0, 0, 0, 0, 0, 1, 2, 3, 0, 0]
    assert_array_equal(constant, expected_constant)
```
As Johannes said `C` is not a valid parameter value (see https://github.com/scikit-image/scikit-image/blob/master/skimage/_shared/interpolation.pxd#L335-L365).

**Comment by sciunto** [comment]
either we remove the test or we raise a ValueError...

**Comment by soupault** [comment]
I move this to 0.14, as it is not critical. To be merged after #2454 .

**Comment by sciunto** [comment]
big :+1
